pkg/kds/v2/server/status_tracker.go

```diff
@@ -98,6 +98,9 @@ func (c *statusTracker) OnDeltaStreamClosed(streamID int64, _ *envoy_core.Node)
 	defer c.mu.Unlock()
 
 	state := c.streams[streamID]
+	if state == nil {
```

Comment:
can the same thing happen in dataplane status tracker https://github.com/kumahq/kuma/blob/f80621987afd7e01fb1ff3b65757014a4fbebde5/pkg/xds/server/callbacks/dataplane_status_tracker.go#L94 ? if so should we also add a guard there?

Reply:
Let me add a guard and logs in all places because it does seem odd

Reply:
Done!